cli/bench/lsp.rs

```diff
@@ -0,0 +1,274 @@
+    let msg = format!(
+      "Content-Length: {}\r\n\r\n{}",
+      value.to_string().as_bytes().len(),
+      value
+    );
+    self.writer.write_all(msg.as_bytes()).await?;
+    self.writer.flush().await.map_err(|e| e.into())
+  }
+}
+
+pub(crate) async fn benchmarks(
+  deno_exe: &PathBuf,
+) -> Result<HashMap<String, u128>, AnyError> {
+  let mut client = LspClient::new(deno_exe)?;
+
+  client
+    .write(json!({
+      "jsonrpc": "2.0",
+      "id": 1,
+      "method": "initialize",
+      "params": {
+        "processId": 0,
+        "clientInfo": {
+          "name": "test-harness",
+          "version": "1.0.0"
+        },
+        "rootUri": null,
+        "initializationOptions": {
+          "enable": true,
+          "codeLens": {
+            "implementations": true,
+            "references": true
+          },
+          "lint": true,
+          "importMap": null,
+          "unstable": false
+        },
+        "capabilities": {
+          "textDocument": {
+            "codeAction": {
+              "codeActionLiteralSupport": {
+                "codeActionKind": {
+                  "valueSet": [
+                    "quickfix"
+                  ]
+                }
+              },
+              "isPreferredSupport": true,
+              "dataSupport": true,
+              "resolveSupport": {
+                "properties": [
+                  "edit"
+                ]
+              }
+            },
+            "synchronization": {
+              "dynamicRegistration": true,
+              "willSave": true,
+              "willSaveWaitUntil": true,
+              "didSave": true
+            }
+          }
+        }
+      }
+    }))
+    .await?;
+
+  let response: LspResponse = client.read().await?;
+  assert!(response.error.is_none());
+
+  client
+    .write(json!({
+      "jsonrpc": "2.0",
+      "method": "initialized",
+      "params": {}
+    }))
+    .await?;
+
+  client
+    .write(json!({
+      "jsonrpc": "2.0",
+      "method": "textDocument/didOpen",
+      "params": {
+        "textDocument": {
+          "uri": "file:///a/file.ts",
+          "languageId": "typescript",
+          "version": 1,
+          "text": "console.log(Deno.args);\n"
+        }
+      }
+    }))
+    .await?;
+
+  let notification: LspNotification = client.read().await?;
+  assert_eq!(notification.method, "textDocument/publishDiagnostics");
+
+  client
+    .write(json!({
+      "jsonrpc": "2.0",
+      "id": 2,
+      "method": "shutdown",
+      "params": null
+    }))
+    .await?;
+
+  let response: LspResponse = client.read().await?;
+  assert!(response.error.is_none());
+
+  client
+    .write(json!({
+      "jsonrpc": "2.0",
+      "method": "exit",
+      "params": null
+    }))
+    .await?;
+
+  let mut exec_times = HashMap::new();
+  exec_times.insert("startup_shutdown".to_string(), client.duration());
+
+  println!("lsp bench: {:?}", exec_times);
+
+  Ok(exec_times)
+}
```

Comment:
Nice.

I have no idea how long this takes, but if its very fast, maybe loop and do this 1000 times (or whatever).

Reply:
About 400ms on my machine and it seems fairly consistent... because there is time to start up the tsc isolate and generate the first diagnostics...  Something like 10 times for this one, but I think this will truly be the "baseline" one, and then we need a "more complex" one.